test/functional/Jsr292/src/com/ibm/j9/jsr292/FilterArgumentsWithCombinerTest.java

```diff
@@ -0,0 +1,435 @@
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package com.ibm.j9.jsr292;
+
+import org.testng.annotations.Test;
+import org.testng.AssertJUnit;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodHandles.Lookup;
+import java.lang.invoke.MethodType;
+import java.lang.invoke.WrongMethodTypeException;
+import java.lang.reflect.Method;
+
+/* Tests MethodHandles.filterArgumentsWithCombiner() with the following FieldTypes for (Parameter/Return Descriptors): Z, B, S, C, I, F, J, D, Ljava/lang/Double;
+ *
+ * For each FieldType, MethodHandles.foldArguments() is tested for boundary and intermediate conditions. Combiner placement and argument indices are also verified.
+ */
+
+ public class FilterArgumentsWithCombinerTest {
+    /* test values */
+    private final static byte constByte1 = (byte) 10;
+    private final static byte constByte2 = (byte) -10;
+    private final static short constShort1 = (short) -12096;
+    private final static short constShort2 = (short) -21510;
+    private final static char constChar1 = (char) 31911;
+    private final static char constChar2 = (char) 3763;
+    private final static int constInt1 = (int) 1213988863;
+    private final static int constInt2 = (int) -1993736192;
+    private final static long constLong1 = (long) 4.21368040135852e+018;
+    private final static long constLong2 = (long) 1.26663739519795e+018;
+    private final static float constFloat1 = (float) 6.12483306976318e+037F;
+    private final static float constFloat2 = (float) 1.53640056404114e+038F;
+    private final static double constDouble1 = (double) 7.33110759312901e+307;
+    private final static double constDouble2 = (double) 1.10819706189633e+307;
+    private final static boolean constBoolMax = true;
+    private final static boolean constBoolMin = false;
+
+    private final static String MHClassName = "java.lang.invoke.MethodHandles";
+    private static Class<?> mhClazz;
+    private static Method fawcMethod;
+
+    static {
+        try {
+            mhClazz = Class.forName(MHClassName);
+            fawcMethod = mhClazz.getDeclaredMethod("filterArgumentsWithCombiner", MethodHandle.class, int.class, MethodHandle.class, int[].class);
+            fawcMethod.setAccessible(true);
+        } catch(Throwable e) {
+            AssertJUnit.fail("FilterArgumentsWithCombinerTest setup failure" + e.getMessage());
+        }
+    }
+
+    /* test filterArgumentsWithCombiner for data type byte */
+    @Test(groups = { "level.extended" })
+	public void test_filterArgumentsWithCombiner_Byte() throws WrongMethodTypeException, Throwable {
+        MethodHandle result;
```

Comment:
Better declare when using it in the test.
e.g. MethodHandle  result = (MethodHandle)fawcMethod.invoke(...)
      result = ...